10up/classifai
# OpenAI image generation tweaks

### Description of the Change

This PR refines and fixes some issues with the OpenAI image generation feature that was added in #419.

1. Ensure the CSS we need is always enqueued. Previously our CSS was in the `language-processing.scss` file but that is only enqueued if the IBM Watson integration is on. So if that feature was off, the CSS we need was never loaded, resulting in some UI issues. In addition to ensuring CSS is always enqueued, moved the media modal CSS into it's own file
2. Changed the prompt input from a normal text input into a textarea, allowing the prompt to be longer without cutting off
3. After images are generated, keep the prompt text in the input so it can easily be tweaked if another request needs to be made
4. Adjust some styling, most based on comments left on #419 after it was merged but also to reduce the size of the generated images so we get 4 across instead of 3 across
5. Add the prompt text as the alt text to any images that we import
6. Remove the secondary import button after an image is imported, as it's no longer needed
7. Moves the `Generate images` tab as the second item, before the `Media Library` tab

Partially closes #440 

![Screen Shot 2023-04-26 at 11 25 09 AM](https://user-images.githubusercontent.com/916738/234672249-d6c85a85-8e8a-4026-b9cb-c6d22a081855.png)

### How to test the Change

1. Setup the OpenAI DALLE image generation feature
2. Within a post, add a new Image block and then enter the media modal flow
3. Switch to the Generate image tab and ensure styling looks correct
4. Enter in a prompt and submit that, ensuring loading states show and images populate
5. Import one or more images, ensuring alt text is set

### Changelog Entry

> Added - Add the image generation prompt as `alt` text to imported generated images
> Changed - Tweaks to the image generation UI
> Fixed - Ensure proper CSS is always loaded based on the enabled features

### Credits

Props @dkotter, @jeffpaul, @mehidi258 

### Checklist:
